.github/workflows/Basic.yml

```diff
@@ -0,0 +1,29 @@
+name: Basic CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        mpi: [ ON, OFF ] 
+        compiler: [gcc, clang]
+        BUILD_TYPE : [Debug, Release]
+    steps:
+    - uses: actions/checkout@v2 
+    - name: blaslapack
+      run: sudo apt install libblas-dev liblapack-dev libopenmpi-dev
+    - name: configure
+      env:
+        CC: ${{ matrix.compiler }}
```

Comment:
If we are building with mpi, then, as far as I remember, you should use the `mpicc` and `mpicxx` wrappers around the compiler. So is this correct?

Reply:
good question I have no idea.

Reply:
according to https://stackoverflow.com/questions/23163075/how-to-compile-an-mpi-included-c-program-using-cmake it should work without it because cmake should take care of it.